src/app/services/keybinding.service.ts

```diff
@@ -0,0 +1,156 @@
+
+
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+  
+  SPDX-License-Identifier: EPL-2.0
+  
+  Copyright Contributors to the Zowe Project.
+*/
+
+import { Injectable } from '@angular/core';
+
+import { Subject, Observable } from 'rxjs/Rx';
+
+@Injectable()
+export class KeybindingService {
+  public keyupEvent: Subject<KeyboardEvent>;
+  public keydownEvent: Subject<KeyboardEvent>;
+
+  constructor() {
+    this.keyupEvent = new Subject();
+    this.keydownEvent = new Subject();
+  }
+
+  registerKeyUpEvent(appChild:Element) {
+    let elm = appChild.closest('div.window');
+    const observable = Observable.fromEvent(elm, 'keyup' ) as Observable<KeyboardEvent>;
+    observable
+      .filter(value => value.altKey)
+      .subscribe(this.keyupEvent);
+  }
+
+  registerKeyDownEvent(appChild:Element) {
+    let elm = appChild.closest('div.window');
+    const observable = Observable.fromEvent(elm, 'keydown' ) as Observable<KeyboardEvent>;
+    observable
+      .filter(value => value.altKey)
+      .subscribe(this.keydownEvent);
+  }
+}
+
+export enum KeyCode {
```

Comment:
You could find keycodes in `@angular/cdk/keycodes`

Reply:
It looks like the Angular keycode is deprecated and it is recommended to not use them https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode
Given that my way is also how Editor currently does things, I think it's okay to keep it like that, so we can control the codes & change them later in case they don't correspond correctly in a certain browser